src/engraving/libmscore/engravingitem.h

```diff
@@ -563,6 +563,9 @@ class EngravingItem : public EngravingObject
     std::vector<Spanner*>& startingSpanners() { return _startingSpanners; }
     std::vector<Spanner*>& endingSpanners() { return _endingSpanners; }
 
+    bool allowRemoveWhenRemovingStaves(staff_idx_t startStaff, staff_idx_t endStaff = 0) const;
+    bool moveDownWhenAddingStaves(staff_idx_t startStaff, staff_idx_t endStaff = 0) const;
```

Comment:
It's not quite clear to me what happens if startStaff == 0 and endStaff == 0. I guess the default value for endStaff should be mu::nidx, right?

Reply:
Actually I need that, cause I wanted to make it possible for the user of this function to just provide the startStaff (if they want to delete only one staff). By setting the default to endStaff = 0, I can distinguish by an invalid input (endStaff == mu::nidx, in which case I exit early) and a default initialization (in which case I just set it to one-past the startStaff). Hope this makes sense

Reply:
>  wanted to make it possible for the user of this function to just provide the startStaff (if they want to delete only one staff)

My concern is what happens if the user wants to delete **only the first staff** in the score (so startStaff == 0 and endStaff == 0). It seems that if we called these functions as follows (we wanted to check **only the first staff**, not the whole score):

`allowRemoveWhenRemovingStaves(0)
`

We would get this result (not what we expected):

`allowRemoveWhenRemovingStaves(0, score()->nstaves() - 1)
`

Am I right?

Reply:
No in that case the function reaches [this check](https://github.com/musescore/MuseScore/pull/15938/files#diff-571f2ae508434dfd554cfdbcfb0427062e955fd81490dec44d09259917d7f562R1166-R1168), and if endStaff == 0 we set it to startStaff + 1. So, in the end, calling allowRemoveWhenRemovingStaves(0) is equivalent to calling allowRemoveWhenRemovingStaves(0, 1). Which is what we want, cause endStaff is assumed to be the index one-past the staff we want to remove